### PR TITLE
games-emulation/dolphin-9999: use sys-libs/minizip-ng and disable steam compile switch

### PR DESCRIPTION
**Pull request by mazes-80**
sys-libs/minizip-ng is still a draft as it may lack some configure switches

P.S.: Sorry about pr #25812, I didn't follow a good workflow (no branch)

### CODE REVIEW

File: sys-libs/minizip-ng/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<!-- maintainer-needed -->
```
Comment (mgorny) [inline sys-libs/minizip-ng/metadata.xml:4]: No can do. New packages must have a dedicated maintained. Try @gentoo/base-system, they already maintain zlib-ng.

File: sys-libs/minizip-ng/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	<upstream>
+		<remote-id type="github">inikep/lz5</remote-id>
```
Comment (mgorny) [inline sys-libs/minizip-ng/metadata.xml:6]: This seems… wrong?

File: sys-libs/minizip-ng/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	</upstream>
+	<use>
+		<flag name=compat>remain compatible with <pkg>sys-libs/zlib[minizip]</pkg></flag>
```
Comment (mgorny) [inline sys-libs/minizip-ng/metadata.xml:9]: "Compatible" how?

File: sys-libs/minizip-ng/minizip-ng-3.0.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Fork of the popular zip manipulation library found in the zlib distribution"
+HOMEPAGE="https://github.com/zlib-ng/minizip-ng"
+SRC_URI="https://github.com/zlib-ng/minizip-ng/archive/refs/tags/3.0.6.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline sys-libs/minizip-ng/minizip-ng-3.0.6.ebuild:10]:
```suggestion
SRC_URI="https://github.com/zlib-ng/minizip-ng/archive/${PV}.tar.gz -> ${P}.tar.gz"
```

File: sys-libs/minizip-ng/minizip-ng-3.0.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+LICENSE="ZLIB"
+SLOT="0"
+KEYWORDS="~amd64"
+
+IUSE="+compat test"
+
+DEPEND="app-arch/bzip2
```
Comment (mgorny) [inline sys-libs/minizip-ng/minizip-ng-3.0.6.ebuild:18]: I'm surprised it doesn't depend on zlib or zlib-ng. Does it bundle it?

**Review by mgorny — CHANGES_REQUESTED** [review]

File: sys-libs/minizip-ng/minizip-ng-3.0.6.ebuild (hunk 5 above)
Comment (mazes-80) [reply sys-libs/minizip-ng/minizip-ng-3.0.6.ebuild:18]: I'm not sure whether it is abi compatible or not.
That's why I described loosely, however it is api compatible, so I just updated description in this way.
Anyway this ebuild still is a draft, I just made it quick to be usable by the package I'm targeting.

**Comment by thesamesam** [comment]
Only Dolphin needs handling now.

```
commit c8c393787952621be914bf8fbbfd725cee6917be
Author: Sam James <sam@gentoo.org>
Date:   Sun Oct 2 02:00:40 2022 +0100

    profiles/base: mask sys-libs/minizip-ng[compat] too

    Signed-off-by: Sam James <sam@gentoo.org>

commit 3441ab9985c98c9f21bf4e529c54e0db84c19187
Author: Sam James <sam@gentoo.org>
Date:   Sun Oct 2 01:58:20 2022 +0100

    sys-libs/minizip-ng: new package, add 3.0.6

    Signed-off-by: Sam James <sam@gentoo.org>

commit 239f765da8dc3e9cd6abe28a89e2558f857505fa
Author: Sam James <sam@gentoo.org>
Date:   Sun Oct 2 01:13:25 2022 +0100

    profiles/base: mask sys-libs/zlib-ng[compat]

    USE=compat / possibly allowing sys-libs/zlib needs to be figured out,
    but packages want to use the actual zlib-ng library, so we need to
    mask it and can't hold the whole thing up on the compat effort.

    Signed-off-by: Sam James <sam@gentoo.org>
```

**Pull request closed without merging**